public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts

```diff
@@ -64,6 +69,45 @@ export default class ResourcePickerData extends DataSourceWithBackend<AzureMonit
     return addResources(rows, parentRow.uri, nestedRows);
   }
 
+  search = async (searchPhrase: string, searchType: 'logs' | 'metrics'): Promise<ResourceRowGroup> => {
+    const searchQuery = {
+      metrics: `
+        resources
+        | where id contains "${searchPhrase}"
+        | where type in (${this.supportedMetricNamespaces.map((ns) => `"${ns.toLowerCase()}"`).join(',')})
+        | order by tolower(name) asc
+        | limit 200
+      `,
+      logs: `
+        resources
+        | union resourcecontainers 
+        | where id contains "${searchPhrase}"
+        | where type in (${logsSupportedResourceTypesKusto})
+        | order by tolower(name) asc
+        | limit 200
+      `,
+    };
+    const { data: response } = await this.makeResourceGraphRequest<RawAzureResourceItem[]>(searchQuery[searchType]);
+    return response.map((item) => {
+      const parsedUri = parseResourceURI(item.id);
+      if (!parsedUri || !(parsedUri.resource || parsedUri.resourceGroup || parsedUri.subscriptionID)) {
+        throw new Error('unable to fetch resource details');
+      }
+      return {
+        name: item.name,
+        id: parsedUri.resource || parsedUri.resourceGroup || parsedUri.subscriptionID,
+        uri: item.id,
+        resourceGroupName: item.resourceGroup,
+        type:
+          (parsedUri.resource && ResourceRowType.Resource) ||
+          (parsedUri.resourceGroup && ResourceRowType.ResourceGroup) ||
+          ResourceRowType.Subscription,
+        typeLabel: resourceTypeDisplayNames[item.type] || item.type,
+        location: locationDisplayNames[item.location] || item.location,
+      };
```

Comment:
nit: I find this a bit more readable but up to you

```suggestion
      let id = parsedUri.subscriptionID;
      let type = ResourceRowType.Subscription;
      if (parsedUri.resource) {
        id = parsedUri.resource;
        type = ResourceRowType.Resource;
      } else if (parsedUri.resourceGroup) {
        id = parsedUri.resourceGroup;
        type = ResourceRowType.ResourceGroup;
      }
      return {
        name: item.name,
        id,
        uri: item.id,
        resourceGroupName: item.resourceGroup,
        type,
        typeLabel: resourceTypeDisplayNames[item.type] || item.type,
        location: locationDisplayNames[item.location] || item.location,
      };
```